src/org/armedbear/lisp/StreamError.java

```diff
@@ -161,9 +161,13 @@ public String getMessage()
         @Override
         public LispObject execute(LispObject arg)
         {
-            if (arg instanceof StreamError)
-                return ((StreamError)arg).getStream();
-            return type_error(arg, Symbol.STREAM_ERROR);
+            final StandardObject obj;
```

Comment:
It should still test whether the arg is a stream-error or a subclass, and signal a type-error otherwise. Here, it will signal an error about an inexistant slot instead.

Reply:
Right -- which is the same error that `type-error-datum` returns on a non-`type-error`:
```lisp
CL-USER(1): (make-condition 'condition)
#<CONDITION {7D9FBAFD}>
CL-USER(2): (type-error-datum *)
#<THREAD "interpreter" {657A43BD}>: Debugger invoked on condition of type SIMPLE-ERROR
  The slot SYSTEM::DATUM is missing from the class #<STANDARD-CLASS CONDITION {57A2A10B}>.

CL-USER(4): (stream-error-stream *)
#<THREAD "interpreter" {657A43BD}>: Debugger invoked on condition of type SIMPLE-ERROR
  The slot STREAM is missing from the class #<STANDARD-CLASS CONDITION {57A2A10B}>.
```

For the record, SBCL appears to handle this the same way (while they strictly check the object is type condition, it doesn't check whether it's typep to stream-error or type-error):
```lisp
(make-condition 'condition)
#<CONDITION {1004CB53D3}>
(type-error-datum *)

debugger invoked on a SIMPLE-ERROR in thread
#<THREAD "main thread" RUNNING {1001558173}>:
  There is no slot named SB-KERNEL::DATUM in #<SB-KERNEL::CONDITION-CLASSOID CONDITION>.

(stream-error-stream *)

debugger invoked on a SIMPLE-ERROR in thread
#<THREAD "main thread" RUNNING {1001558173}>:
  There is no slot named STREAM in #<SB-KERNEL::CONDITION-CLASSOID CONDITION>.
```

In any case, an error is an error -- this should be satisfactory.

Reply:
It's not specified to be a type-error in the ANSI spec (no exceptional situations were listed) but if will probably be in [WSCL](https://github.com/s-expressionists/wscl/blob/main/wscl-issues/proposed/stream-error-stream-type-error) for what it is worth.

Reply:
That's fair -- it looks like WSCL has a mirror proposal for type-error-datum too, so I'll make a separate commit to change the error for both functions.

Reply:
Awesome.

I have PRs prepped to update the current practices for stream-error-stream and others in WSCL. I'll put them on pause until your PR gets resolved.

Reply:
I've added the changes, thanks for the feedback!

```lisp
CL-USER(3): (make-instance 'standard-object)
#<STANDARD-OBJECT {659BC8F9}>
CL-USER(4): (stream-error-stream *)
#<THREAD "interpreter" {26AA632D}>: Debugger invoked on condition of type TYPE-ERROR
  The value #<STANDARD-OBJECT {659BC8F9}> is not of type STREAM-ERROR.

CL-USER(6): (type-error-datum *)
#<THREAD "interpreter" {26AA632D}>: Debugger invoked on condition of type TYPE-ERROR
  The value #<STANDARD-OBJECT {659BC8F9}> is not of type TYPE-ERROR.
```

I'll go ahead and resolve this thread